MH4U Dex/CoreDataController.m

```diff
@@ -177,6 +210,50 @@ - (void)loadMonsterDropData
     }
 }
 
+- (void)loadQuestData
+{
+    NSArray *questList = [self loadArrayFromJsonFileNamed:QuestsFileName];
+    
+    for (NSDictionary *questDict in questList) {
+        Quest *quest;
+        NSNumber *questID = questDict[QuestIDKey];
+        NSPredicate *questPredicate = [NSPredicate predicateWithFormat:@"%K == %d", QuestAttributes.id, questID.intValue];
+        quest = (Quest *)[self getUniqueEntityWithEntityName:[Quest entityName] withPredicate:questPredicate];
+        //Only add the quest if it is not already there
+        if (!quest) {
+            // Quests have a lot of information to populate, so I'm moving this to a helper method to cut down on the size of this method.
+            quest = [self questFromDictionary:questDict];
+        }
+    }
+    if (![self attemptSaveContext]) {
+        NSLog(@"WARNING: Failed to save the item context!");
+    }
+}
+
+- (void)loadQuestDropData
+{
+    NSArray *questDropList = [self loadArrayFromJsonFileNamed:QuestDropsFileName];
+    
+    for (NSDictionary *dropDict in questDropList) {
+        QuestDrop *drop;
+        NSNumber *dropID = dropDict[QuestDropIDKey];
+        NSPredicate *dropPredicate = [NSPredicate predicateWithFormat:@"%K == %d", QuestDropAttributes.id, dropID.intValue];
+        drop = (QuestDrop *)[self getUniqueEntityWithEntityName:[QuestDrop entityName] withPredicate:dropPredicate];
+        if (!drop) {
+            drop = [self questDropFromDictionary:dropDict];
+            if (!drop) {
+                //If we reach this point, then the quest that this drop belongs to did not exist, and we should return without saving the context;
```

Comment:
If I'm reading it correctly, it's not just that it did not exist, it's that you tried to create it and failed to do so...